src/aof.c

```diff
@@ -265,7 +417,45 @@ void flushAppendOnlyFile(int force) {
      * or alike */
 
     latencyStartMonitor(latency);
-    nwritten = write(server.aof_fd,server.aof_buf,sdslen(server.aof_buf));
+#ifdef USE_NVML
+    if (server.aof_use_nvml == 1) {
+        size_t len = sdslen(server.aof_buf);
+        size_t newlen = server.aof_current_size + len;
+
+        /* Check if there's required space, if not try to rewrite first */
+        if (newlen > server.aof_nvml_log_size)
+        {
+            /* if there's not enough space try to compact AOF file */
+            redisLog(REDIS_ERR, "Not enough space to write AOF file");
+            redisLog(REDIS_NOTICE, "Starting background rewrite of AOF file");
+            if (rewriteAppendOnlyFileBackground() != REDIS_OK) {
+                redisLog(REDIS_ERR, "Starting background rewrite of AOF file failed!");
+            }
+            nwritten = -1;
+            errno = ENOSPC;
+        } else {
+            if (!pmemlog_append(server.aof_plp, server.aof_buf, len)) {
+                nwritten = len;
+
+                /* if reached threshold for AOF rewrite it */
+                if (server.aof_rewrite_perc > 0 &&
+                        server.aof_current_size > (server.aof_nvml_log_size *
```

Comment:
The original meaning of `aof_rewrite_perc` is to trigger AOF rewrite when the current AOF size is N% of the AOF size after the last rewrite. By default it's set to 100%, so rewrite is triggered every time AOF doubles its size (since the last rewrite).
We would like to trigger rewrite also when AOF size reaches i.e. 80% of the pmemlog pool size. But if we reuse the same variable, then it affects the algorithm of the original rewrite trigger, and AOF rewrites would be triggered much often. For instance, if you set it to 50%, then rewrite will be triggered with 2x higher frequency.
Possible solutions:
1) Use another config variable for this trigger (preferred)
2) When in NVML mode, disable the original trigger, and leave the new one only - this must be clearly explained in default redis.conf.
3) Heuristic trigger - every time the rewrite is completed, save the size of the AOF diff. The next rewrite must be triggered when the free space in current AOF can still hold the diff (+ some safety margin).